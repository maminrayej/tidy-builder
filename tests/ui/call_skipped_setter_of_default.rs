#[derive(tidy_builder::Builder)]
struct MyStruct {
    #[builder(skip)]
    set_later_opt: Option<usize>,

    #[builder(skip)]
    #[builder(default = 0)]
    set_later_def: usize,
}

fn main() {
    let _ = MyStruct::builder().set_later_def();
}
