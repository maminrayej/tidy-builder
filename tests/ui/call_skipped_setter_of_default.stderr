error[E0599]: no method named `set_later_def` found for struct `MyStructBuilder` in the current scope
  --> tests/ui/call_skipped_setter_of_default.rs:12:33
   |
1  | #[derive(tidy_builder::Builder)]
   |          --------------------- method `set_later_def` not found for this
...
12 |     let _ = MyStruct::builder().set_later_def();
   |                                 ^^^^^^^^^^^^^-- help: remove the arguments
   |                                 |
   |                                 field, not a method
